src/plugins/intel_cpu/src/config.cpp

```diff
@@ -73,192 +68,248 @@ void Config::applyDebugCapsProperties() {
 void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
     const auto streamExecutorConfigKeys =
         streamExecutorConfig.get_property(ov::supported_properties.name()).as<std::vector<std::string>>();
-    const auto hintsConfigKeys = perfHintsConfig.SupportedKeys();
     for (const auto& kvp : prop) {
         const auto& key = kvp.first;
-        const auto& val = kvp.second.as<std::string>();
-        IE_SUPPRESS_DEPRECATED_START
+        const auto& val = kvp.second;
         if (streamExecutorConfigKeys.end() !=
             std::find(std::begin(streamExecutorConfigKeys), std::end(streamExecutorConfigKeys), key)) {
-            streamExecutorConfig.set_property(key, val);
+            streamExecutorConfig.set_property(key, val.as<std::string>());
             if (key == ov::affinity.name()) {
-                const auto affinity_val = ov::util::from_string(val, ov::affinity);
-                if (affinity_val == ov::Affinity::CORE || affinity_val == ov::Affinity::HYBRID_AWARE) {
-                    enableCpuPinning = true;
-                    changedCpuPinning = true;
-                } else if (affinity_val == ov::Affinity::NUMA) {
-                    enableCpuPinning = false;
-                    changedCpuPinning = true;
+                changedCpuPinning = true;
+                try {
+                    const auto affinity_val = val.as<ov::Affinity>();
+                    enableCpuPinning =
+                        (affinity_val == ov::Affinity::CORE || affinity_val == ov::Affinity::HYBRID_AWARE) ? true
+                                                                                                           : false;
+                } catch (const std::exception&) {
+                    OPENVINO_THROW("Wrong value ",
+                                   val.as<std::string>(),
+                                   "for property key ",
+                                   key,
+                                   ". Expected only CORE/NUMA/HYBRID_AWARE.");
                 }
             }
-        } else if (hintsConfigKeys.end() != std::find(hintsConfigKeys.begin(), hintsConfigKeys.end(), key)) {
-            perfHintsConfig.SetConfig(key, val);
+        } else if (key == ov::hint::performance_mode.name()) {
+            try {
+                hintPerfMode = val.as<ov::hint::PerformanceMode>();
```

Comment:
if been called with 2.0 val here should already be type of ov::hint::PerformanceMode, if not,  here will always throw, because  you cannot convert string directly to this enum type.
so the question is, or maybe with the general logic in this config, most of exceptions may not gonna happen here maybe?

Reply:
Maybe you have made a mistake. Because all `ov::Any` with `std::string` can convert to specified data type if the data type have defined its conversion rule, such as `ov::hint::PerformanceMode`:

https://github.com/openvinotoolkit/openvino/blob/89494ab412092761d3bc8cdb36d64cbf675bc6e5/src/inference/include/openvino/runtime/properties.hpp#L328-L345

We can write a simple test case for it:
```
    std::string perf_mode_str="THROUGHPUT";
    ov::Any perf_mode_any = perf_mode_str;
    auto perf_mode = perf_mode_any.as<ov::hint::PerformanceMode>();
    std::cout << "perf_mode = " << perf_mode << std::endl;
```

The test result is below without any exception:

` perf_mode = THROUGHPUT`

So above `as<actual_type>()` has no any problem.

Reply:
ah..yes, you are right

Reply:
then maybe the exception is ov exception here for this property..?
also seems a lot of repeated type of try catch blocks... maybe we can consider optimize this?

Reply:
`as<>()` is not always successful, for example for an invalid value/string it will throw exception, we should catch such problem.
`std::exception` is base class of 'ov::exception`, seems there is not too much different?

Reply:
not sure , we can merge first if needed. I remember previously we once working on remove not needed catch (...) or std::exception, if only ov::Exception is expected here.  cc @ilya-lavrenov

Reply:
ok, let's use ov::exception.

Reply:
I agree that it's better to drop exceptions in plugin in future, and rely on operator >> which parses enum from string